lib/cn/kvset_split.c

```diff
@@ -0,0 +1,652 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+/*
+ * Copyright (C) 2022 Micron Technology, Inc.  All rights reserved.
+ */
+
+#include <hse_util/event_counter.h>
+#include <hse_util/assert.h>
+#include <hse_util/keycmp.h>
+#include <hse_util/logging.h>
+
+#include <hse/limits.h>
+
+#include <hse_ikvdb/cn.h>
+#include <hse_ikvdb/ikvdb.h>
+#include <hse_ikvdb/cndb.h>
+#include <hse_ikvdb/kvset_builder.h>
+
+#include <mpool/mpool.h>
+
+#include "kvs_mblk_desc.h"
+#include "kvset.h"
+#include "kvset_split.h"
+#include "kvset_internal.h"
+#include "kblock_builder.h"
+#include "kblock_reader.h"
+#include "vblock_reader.h"
+#include "wbt_reader.h"
+#include "omf.h"
+#include "cn_tree.h"
+#include "cn_tree_internal.h"
+
+/**
+ * struct kvset_split_work - work struct for kvset split
+ */
+enum { LEFT = 0, RIGHT = 1 };
+
+struct kvset_split_work {
+    struct hlog           *hlog;    /* composite hlog */
```

Comment:
Hmm, I still think you could use an array encompassing all 3 components instead of an array per element.